### ISSUE TITLE
Retry during reroute after table mode change leads to excess rules

### ISSUE THREAD

**Issue by rtretyak** [telstra/open-kilda#3941]
Case looks similar to #3940

1. A flow in single-table mode, on single-table mode switches
2. Src switch mode is changed to multi-table (flow remains in single-table mode until next reroute)
3. Current path breaks (ISL down) and reroute happens
4. During reroute, system for some reason is unable to install rules on new path from the first try, start retries and one of the retries is successful 
5. Run switch validation for related switches
**Expected:** no rule discrepancies on switches
**Actual:** Src switch has some excess rules
```
"excess": [
            2305843009213795558,
            4611686018427489510
        ]
```
- Also in Kibana errors are present, similar to #3940: `segment and flow level multi-table flag values are incompatible to each other - flow(true) != segment(false)"`, but flow can be successfully deleted, unlike 3940

**Comment by rtretyak** [comment]
autotest available

**Comment by rtretyak** [comment]
Closed by #3946
